neovim/api/common.py

```diff
@@ -60,10 +60,26 @@ def __init__(self, obj, api_prefix):
         self._obj = obj
         self._api_prefix = api_prefix
 
+        self._api_names = None
+
     def __getattr__(self, name):
         """Return wrapper to named api method."""
         return functools.partial(self._obj.request, self._api_prefix + name)
 
+    def __dir__(self):
+        """Return info via nvim_get_api_info.
+
+        This can be used for introspection, and especially completion.
+        """
+        if self._api_names is None:
+            api_info = self._obj.request('nvim_get_api_info')
```

Comment:
there is a shared metadata object as `nvim.metadata` though it is a bit tricky to access it as `_obj` could be either a `Nvim` or a `Remote`. Guess you could do something like `nvim = _obj._session if isinstance(obj,Remote) else _obj`